### ISSUE TITLE
Anonymous classes and Type declarations cause false positive for UndefinedVariable bug

### ISSUE THREAD

**Issue by yokozawa0701** [sirbrillig/phpcs-variable-analysis#332]
Here's a simple example that generates UndefinedVariable for using Anonymous classes and Type declarations.

```php
class Test
{
    public int $id = 1;  // This is no error.

    public function test_1(): object
    {
        return new class
        {
            public int $id = 123456;  // Unused variable $id. (VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable)
        };
    }

    public function test_2(): object
    {
        return new class
        {
            public $id = 123456;  // This is no error.
        };
    }
}
```

I am using v2.11.19.
Thank you.

**Comment by sirbrillig** [comment]
Should be fixed by https://github.com/sirbrillig/phpcs-variable-analysis/pull/337
